include/tsl/ordered_hash.h

```diff
@@ -1098,6 +1098,44 @@ class ordered_hash : private Hash, private KeyEqual {
     return 1;
   }
 
+  /**
+   * Clear a bucket without touching the container holding the values.
+   */
+  void clear_bucket(typename buckets_container_type::iterator it) {
+    tsl_oh_assert(it != m_buckets_data.end());
+    it->clear();
+    backward_shift(std::size_t(std::distance(m_buckets_data.begin(), it)));
+  }
+
+  /**
+   * Remove all entries for which the given predicate matches.
+   */
+  template <class Predicate>
+  size_type erase_if(Predicate &pred) {
+    using const_ref = typename values_container_type::const_reference;
+    auto last = m_values.end();
+    auto first = m_values.begin();
+    for (; first != last; ++first) {
+      if (pred(static_cast<const_ref>(*first))) {
+        clear_bucket(find_key(KeySelect()(*first), hash_key(KeySelect()(*first))));
+        for (auto it = first; ++it != last; ) {
```

Comment:
I would avoid incrementing the iterator during the comparison, it is unintuitive. Something like
`for (auto it = first + 1; it != m_values.end(); ++it) {` is preferable I think.

Reply:
I used `auto it = std::next(first)`.